packages/react-query/src/useIsFetching.ts

```diff
@@ -12,13 +12,27 @@ export function useIsFetching(
   const client = useQueryClient(queryClient)
   const queryCache = client.getQueryCache()
 
-  return React.useSyncExternalStore(
-    React.useCallback(
-      (onStoreChange) =>
-        queryCache.subscribe(notifyManager.batchCalls(onStoreChange)),
-      [queryCache],
-    ),
-    () => client.isFetching(filters),
-    () => client.isFetching(filters),
+  const [number, setNumber] = React.useState<number>(() =>
+    client.isFetching(filters),
   )
+
+  const filtersRef = React.useRef(filters)
+  React.useEffect(() => {
+    filtersRef.current = filters
+  }, [filters])
```

Comment:
`useMutationState()` was memoizing filters, so I'm guessing we should be doing it here too

Reply:
this could be done without dependency array - updating the ref after every render is fine.